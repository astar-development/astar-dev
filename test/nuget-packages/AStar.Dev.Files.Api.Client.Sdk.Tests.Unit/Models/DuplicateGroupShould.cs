using AStar.Dev.Files.Api.Client.SDK.Models;

namespace AStar.Dev.Files.Api.Client.Sdk.Models;

public sealed class DuplicateGroupShould
{
    [Fact]
    public void ReturnTheExpectedToString() =>
        new DuplicateGroup
            {
                Duplicates = new List<DuplicatesDetails>
                             {
                                 new ()
                                 {
                                     DetailsLastUpdated = new (2025, 12, 23) ,
                                     FileName           = "MockFileName",
                                     FileSize           = 1234,
                                     Height             = 345,
                                     Width              = 567,
                                     DirectoryName      = "MockDirectoryName",
                                     FileAccessDetailId = 1,
                                     FileHandle         = "MockFileHandle",
                                     HardDeletePending  = true,
                                     Id                 = 2,
                                     Instances          = 3,
                                     IsImage            = true,
                                     LastViewed         = new DateTime(2025, 12, 24),
                                     MoveRequired       = true,
                                     SoftDeleted        = true,
                                     SoftDeletePending  = true
                                 }
                             },
                FileSize     = 123456,
                FileGrouping = { FileSize = 2345, Height = 1432, Width = 2221 }
            }
            .ToString()
            .ShouldMatchApproved();
}
