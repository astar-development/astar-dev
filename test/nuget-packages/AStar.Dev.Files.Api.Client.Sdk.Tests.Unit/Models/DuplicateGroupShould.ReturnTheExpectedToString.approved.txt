{
  "fileSize": -1,
  "fileSizeForDisplay": "-0.00 Kb",
  "fileGrouping": {
    "fileSize": 0,
    "height": 0,
    "width": 0
  },
  "duplicates": []
}